### PR TITLE
speculos: Fasten graphics handling

### PR DESCRIPTION
**Pull request by xchapron-ledger**
This changes allow to reduce app-boilerplate stax tests run time from 50s to 18s on my setup.
Note that this could be further reduced to 10s if using ragger configuration BACKEND_SCOPE="session".

### CODE REVIEW

**Review by cedelavergne-ledger — APPROVED** [review]

File: speculos/mcu/display.py [hunk 1]
```diff
@@ -99,34 +71,67 @@ class FrameBuffer:
...
+        return iobytes.getvalue()
 
     def take_screenshot(self) -> Tuple[Tuple[int, int], bytes]:
-        self.current_screen_size, self.current_data = self.screenshot.get_image()
-        return self.current_screen_size, self.current_data
+        return self.current_screen_size, self._get_image()
 
     def update_screenshot(self) -> None:
-        self.screenshot.update(self.pixels)
+        self.screenshot_pixels = {**self.screenshot_pixels, **self.pixels}
```
Comment (lpascal-ledger) [inline speculos/mcu/display.py:134]:
```suggestion
        self.screenshot_pixels.update(self.pixels)
```

File: speculos/mcu/vnc.py [hunk 2]
```diff
@@ -48,23 +48,25 @@ def file(self) -> IO[bytes]:
         assert self.subprocess.stdout is not None
         return self.subprocess.stdout
 
-    def redraw(self, pixels):
+    def redraw(self, pixels, default_color):
```
Comment (lpascal-ledger) [inline speculos/mcu/vnc.py:51]: While we're at it!
```suggestion
    def redraw(self, pixels: Dict[Tuple[int, int], int], default_color: int) -> None:
```

**Review by lpascal-ledger — APPROVED** [review]
Nice improvement !

File: speculos/mcu/display.py (hunk 1 above)
Comment (xchapron-ledger) [reply speculos/mcu/display.py:134]: Not sure, but in previous version there was a comment saying:
`# Don't call update, replace the object instead`
I trusted it, without testing the performance...

**Pull request merged**